### ISSUE TITLE
Spyder is sometimes not detecting file changes from external editors

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#836]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2011-11-14T13:43:04Z_

Spyder Version:  2.1.1
Python Version:  2.6.1
Qt Version:      4.7.3, PyQt4 (API v1) 4.8.5 on Windows (XP SP3) 

What steps will reproduce the problem?

 1. Open a file in Spyder
2. Open the file in an external editor
3. Make a change and save it in the external editor
4. Return to Spyder and see that Spyder does not reload the file What is the expected output? What do you see instead? A short term work around would be to have the "Revert" option enabled at all times.  I think this used to work and Spyder would reload files but it doesn't appear to be working anymore.  I frequently use a diff program to merge changes into files from a VCS.  I've started losing work recently because I will merge changes in the Diff application and return to Spyder thinking the file will reload only to find out later it did not reload after I've saved new changes to the file inside Spyder. 

Please provide any additional information below

. I consider this a critical issue.  I can't use an editor that results in losing work.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=836_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-11-15T01:07:37Z_

I agree this is a critical issue but I can't reproduce this behavior on any of machines. 

Is anyone else able to reproduce this bug?

**Labels:** -Priority-Medium Priority-Critical Milestone-Release2.1

**Comment by spyder-issues-migrator** [comment]
_From [techtonik@gmail.com](https://code.google.com/u/techtonik@gmail.com/) on 2011-11-15T08:00:40Z_

Well, I had another problem that Spyder was reloading file automatically losing my changes. To me there should be at least some popup dialog with warning.

**Comment by spyder-issues-migrator** [comment]
_From [techtonik@gmail.com](https://code.google.com/u/techtonik@gmail.com/) on 2011-11-15T08:02:08Z_

Steve, do you see anything in internal console?

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2011-11-17T17:12:42Z_

The issue appears to have resolved itself.  Maybe it was an OS work or
sunbathing...  How does Spyder get notifications that files have changed?

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-11-18T05:26:06Z_

It's using this Qt method: http://doc.qt.nokia.com/latest/qfileinfo.html#lastModified here: https://code.google.com/p/spyderlib/source/browse/spyderlib/widgets/editor.py#1525

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2011-11-20T22:19:28Z_

**Labels:** Cat-Editor

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2011-11-22T09:53:23Z_

I've recreated this issue today.  I've got a file open in spyder and I've checked it out from VCS and I'm not getting any notification that the file has changed.  There's nothing in the internal console.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-11-22T09:59:25Z_

I think I've reproduced it too in the same situation. I'll try to reproduce it again to understand what's going on.

**Summary:** Spyder not detecting file changes from external editors

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2011-12-22T11:23:22Z_

What's the status on this?  I just almost lost some work again but luckily still had the compare app open and was able to resave my merged data after I had saved over it from Spyder...

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-12-23T06:05:58Z_

I didn't have the time to investigate this further and I can't reproduce it.

Would you be able to make a procedure on how to reproduce this bug?

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2012-02-07T11:15:02Z_

I have not figured out a specific way to reproduce this issue, but it continues to persist.  Maybe it's an OS specific issue?  Can we (or at least have an option) to leave the Revert menu item always enabled?  I have to close and re-open the file to get the latest contents of a file.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-02-15T13:45:17Z_

This last proposition seems a good compromise.

And I see that other editors (like SciTE) are never disabling this option, even if file  state is not "modified". So, let's always enable this action.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-02-22T12:52:28Z_

This issue was updated by revision 146510776f3a .

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2012-02-22T13:42:03Z_

Thanks for making this short term fix.  Do you have any ideas for investigating the core issue?  I sure wish I did!

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2012-03-17T07:56:32Z_

Unfortunately, I'm stuck with this issue.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-02-12T12:22:04Z_

**Summary:** Spyder is sometimes not detecting file changes from external editors  \<span class="oldvalue"> (was: Spyder not detecting file changes from external editors) \</span>  
  **Status:** Accepted  
  **Labels:** -Milestone-Release2.1 MS-v2.2

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-02-12T12:26:37Z_

issue #1268 has been merged into this issue.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-02-12T12:29:01Z_

@-contrebasse: are you able to reproduce this issue?
If so, please share a step-by-step procedure here.

**Comment by spyder-issues-migrator** [comment]
_From [steve.f....@gmail.com](https://code.google.com/u/117650558843796505335/) on 2013-02-12T12:31:44Z_

I still haven't found a way to specifically reproduce this issue, but it happens to me on almost daily basis now that I'm using git for version control (constantly swapping files on disk).

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-02-12T12:43:07Z_

Unfortunately, this is the only way to fix this issue: we have to know how to reproduce the chain of events that lead to this bug. Otherwise, I won't be able to fix it.

Moreover, these days, I'm writing code very occasionally...

**Comment by spyder-issues-migrator** [comment]
_From [contrebasse](https://code.google.com/u/contrebasse/) on 2013-02-14T00:15:57Z_

It happens everytime for me, using the exact same steps as the original post. I've never seen the file being reloaded.

Also, where is the revert option int he menus ? I can't find it.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-02-18T07:54:15Z_

I must say that I develop Spyder with Spyder and I haven't seen this issue before. I use mercurial bookmarks and constantly move back and forth between our history too, and Spyder always reloads the changes correctly.

But I just tested the steps from comment `#1` and I can confirm the problem.

**Comment by spyder-issues-migrator** [comment]
_From [philipp....@googlemail.com](https://code.google.com/u/106243277547401574018/) on 2013-02-27T04:39:18Z_

I have this issue all the time. Spyder has never reloaded a file, I have to manually use "revert" to get the updated file. I guess it's because all my files are on an NFS filesystem which seems to be difficult with detecting file changes (Dropbox client for example has to be restartet when the files have been changed on a different machine). 
If I change the file with an editor on the same machine spyder detects the changes. If I change the file on another machine (e.g. because I logged in remotely on the file server) spyder ignores it.

OS: Linux Mint 14 [Ubuntu 12.10 derivative]
Filesystem: ext4 mounted via NFS from remote server

A simple workaround would be a "revert all" button.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-03-14T01:15:13Z_

This issue was updated by revision 242d45e6c68e .

**Status:** Fixed

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-03-14T01:16:18Z_

At last, I have been able to reproduce the issue (was occuring only after creating a new file inside Spyder) and fix it.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2013-03-21T01:10:36Z_

issue #1328 has been merged into this issue.

**Comment by christian-ruf** [comment]
this is still happening
